3rdparty/CMakeLists.txt

```diff
@@ -389,7 +389,62 @@ string(APPEND NBL_IMPL_IMCONFIG_CONTENT
 
 // note we override it with default imguis (void*) type
 #include <cstdint>
-#define ImTextureID uint32_t
+
+//! Custom "ImTextureID" info struct for Nabla UI backend purposes about resource sampler & texture descriptor binding's array indicies
+//! must be 4 bytes size & alignment to pass imgui static asserts (it checks for contiguous blocks in memory to make sure it can do some memcpies)
+struct SImResourceInfo 
+{ 
+	//! texture descriptor binding's array index
+    uint32_t textureID : 26; 
+
+	//! sampler descriptor binding's array index
+	uint32_t samplerIx : 6;
+
+    SImResourceInfo() : textureID(0u), samplerIx(0u) {}
+
+    SImResourceInfo(uint32_t texID) 
+    {
+        textureID = texID;
+        samplerIx = 0u;
+    }
+
+	explicit operator intptr_t() const
+	{
+		return static_cast<intptr_t>(textureID);
+	}
+
+    bool operator==(const SImResourceInfo& other) const
+    {
+        return textureID == other.textureID;
+    }
+
+    bool operator!=(const SImResourceInfo& other) const
+    {
+        return textureID != other.textureID;
+    }
+
+    bool operator<(const SImResourceInfo& other) const
+    {
+        return textureID < other.textureID;
+    }
+
+    bool operator>(const SImResourceInfo& other) const
+    {
+        return textureID > other.textureID;
+    }
+
+    bool operator<=(const SImResourceInfo& other) const
+    {
+        return textureID <= other.textureID;
+    }
+
+    bool operator>=(const SImResourceInfo& other) const
+    {
+        return textureID >= other.textureID;
+    }
+};
```

Comment:
why generate such stuff with CMake when it would just make more sense to place it in a header?

Reply:
resolved, the only valid place (if not autogen) is in the submodule